### DISCUSSION TITLE
Access Component Functions in another class

### DISCUSSION THREAD

**Discussion by sheershak-agarwal** [nasa/fprime#1016] [Q&A] [answered]
Hello,

I want basic inheritance functionality. I have a Class for the state Machine and I want to call the gpios, uart write etc functions of the ComponentImpl in the state machine class(GpsSM). 

I have GpsComponentImpl.hpp, GpsComponentImpl.cpp and GpsSM.h in the same folder.

I am adding this in my GpsSM.h
#include "App/Common/StateMachine.h"
#include "GpsComponentImpl.hpp"

class GpsSM : public StateMachine, **public App:GpsComponentImpl** {};

but I get an error In file included from /home/sheershak/Desktop/F-Prime-Debug/App/Gps/GpsComponentImpl.hpp:17:0,
App/Gps/CMakeFiles/App_Gps.dir/build.make:103: recipe for target 'App/Gps/CMakeFiles/App_Gps.dir/GpsComponentImpl.cpp.obj' failed
                 from /home/sheershak/Desktop/F-Prime-Debug/App/Gps/GpsComponentImpl.cpp:14:
**/home/sheershak/Desktop/F-Prime-Debug/App/Gps/GpsSM.h:17:1: error: expected class-name before ‘{’ token
 {**
 ^

I read some other links regarding the issue but couldn't find solution. I also see there is GpsComponentBase. I tried inheriting that too but same error. Also I am creating an object of GpsSM in GpsComponentImpl.hpp with private member.

**Comment by timcanham** [comment]
That usually means that the compiler can't find the base class types. Double-check your spelling including any namespaces.

**Comment by LeStarch** [comment, marked as answer]
I think you need "::" for your namespace.

```
public App:GpsComponentImpl {}
```

Needs to be

```
public App::GpsComponentImpl {}
```

**Reply by sheershak-agarwal** [reply]
Yes, there was a problem in the logic. I was trying to create an object GpsSM in GpsComponentImpl class and trying to inherit GpsComponentImpl in the GpsSM. I got rid of the GpsSM and added the logic in Impl class. That solves my issue.
Thanks
